stackslib/src/net/dns.rs

```diff
@@ -324,30 +322,27 @@ impl DNSClient {
 
     pub fn poll_lookup(&mut self, host: &str, port: u16) -> Result<Option<DNSResponse>, net_error> {
         let req = DNSRequest::new(host.to_string(), port, 0);
-        if !self.requests.contains_key(&req) {
-            return Err(net_error::LookupError(format!(
-                "No such pending lookup: {}:{}",
-                host, port
-            )));
-        }
+        if let Some(e) = self.requests.get_mut(&req) {
+            let _ = match e {
+                None => {
+                    return Ok(None);
+                }
+                Some(resp) => resp,
+            };
```

Comment:
This can be replaced by
```suggestion
            if e.is_none() {
                return Ok(None);
            };
```